lib/environment.rb

```diff
@@ -164,6 +164,20 @@ def admin_console_url
       return @admin_console_url
     end
 
+    def authentication_url
+      unless @authentication_url
+        if opts[:authentication_url]
+          @authentication_url= opts[:authentication_url]
+        else
+          authenticationproject = Project.new(name: "openshift-authentication", env: self)
+          authenticationservice = Service.new(name: "openshift-authentication", project: authenticationproject )
+          authenticationroute = Route.new(name: "openshift-authentication", project: authenticationproject, service: authenticationservice)
+          @authentication_url = "https://" + authenticationroute.dns(by: admin)
```

Comment:
can't we figure out auth URL by trying to login to API? Sounds like better than requiring admin access.